indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/RecordSupplierInputSourceTest.java

```diff
@@ -135,6 +137,24 @@ public void testReadTimeout() throws IOException
     Assert.assertTrue(supplier.isClosed());
   }
 
+  @Test
+  public void testRecordSupplierInputSourceThrowsSamplerExceptionWhenExceptionDuringSeek()
+  {
+    final RecordSupplier<?, ?, ?> supplier = Mockito.mock(RecordSupplier.class);
+    Mockito.when(supplier.getPartitionIds("test-stream"))
+           .thenThrow(new StreamException(new Exception("Something bad happened")));
+
+    //noinspection ResultOfObjectAllocationIgnored
```

Comment:
Do we need this `noinspection` annotation? My understanding is that the static checks don't run on test code, but I might be wrong.

Reply:
Plus we are using result now, which was the `SamplerException`, unless I am mistaken and this `noinspection` was for something else. Either way, we can fix this later.

Reply:
This is technically complaining that the newly allocated object inside the `assertThrows` closure isn't assigned to anything :) which in this case we don't really care.